samples/Databricks-AzureSQL/DatabricksNotebooks/SQL Spark Connector - Python AAD Auth.py

```diff
@@ -0,0 +1,107 @@
+
+# In the below example, we're using a Databricks utility that facilitates acquiring secrets from 
+# a configured Key Vault.  
+
+# Service Principal Client ID - Created in App Registrations from Azure Portal
+service_principal_id = dbutils.secrets.get("principalClientId")
+
+# Service Principal Secret - Created in App Registrations from Azure Portal
+service_principal_secret = dbutils.secrets.get("principalSecret")
+
+# SQL Server URL
+url = "jdbc:sqlserver://azuresqlserver.database.windows.net"
+
+# Database Name
+database_name = "TestDatabase"
+
+# Database Table Name
+db_table = "dbo.TestTable" 
+
+# Encrypt
+encrypt = "true"
+
+# Host Name in Certificate
+host_name_in_certificate = "*.database.windows.net"
+
+# COMMAND ----------
+
+# DBTITLE 1,Authentication Options
+# SERVICE PRINCIPAL AUTHENTICATION
+#   You will need to obtain an access token.
+#   The "accessToken" option is used in the spark dataframe to indicate this 
+#   authentication modality.
+context = adal.AuthenticationContext(authority)
+token = context.acquire_token_with_client_credentials(resource_app_id_url, service_principal_id, service_principal_secret)
+
+# Set Access Token
+access_token = token["accessToken"]
+
+# ACTIVE DIRECTORY PASSWORD AUTHENTICATION
+#   The "authentication" option with the value of "ActiveDirectoryPassword"
+#   is used in the spark dataframe to indicate this 
+#   authentication modality. 
+#
+#   The "user" and "password" options apply to both SQL Authentication
+#   and Active Directory Authentication.  SQL Authentication is used
+#   by default and can be switched to Active Directory with the
+#   authentication option above.
+user = dbutils.secrets.get("adUser")
+password = dbutils.secrets.get("adPassword")
+
+
+# COMMAND ----------
+
+# DBTITLE 1,Query SQL using Spark with Service Principal Token
+jdbc_df = spark.read.format("com.microsoft.sqlserver.jdbc.spark") \
+              .option("url", url) \
+              .option("dbtable", db_table) \
+              .option("accessToken", access_token) \
+              .option("encrypt", encrypt) \
+              .option("databaseName", database_name) \
+              .option("hostNameInCertificate", host_name_in_certificate) \
+              .load() 
+
+display(jdbc_df.select("SourceViewName").limit(1))
+
+# COMMAND ----------
+
+# DBTITLE 1,Query SQL using Spark with Active Directory Password
+jdbc_df = spark.read.format("com.microsoft.sqlserver.jdbc.spark") \
+              .option("url", url) \
+              .option("dbtable", db_table) \
+              .option("authentication", "ActiveDirectoryPassword") \
+              .option("user", user) \
+              .option("password", password) \
+              .option("encrypt", encrypt) \
+              .option("databaseName", database_name) \
+              .option("hostNameInCertificate", host_name_in_certificate) \
+              .load() 
+
+display(jdbc_df.select("SourceViewName").limit(1))
```

Comment:
Can we make this simply `display(jdbc_df)`? I know it is self-apparent, but considering this is a fairly neutral example, it may throw off an less-initiated user if `SourceViewName` is not a column in their source table.